pkg/authorize/authorizer.go

```diff
@@ -0,0 +1,55 @@
+package authorize
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/dgrijalva/jwt-go"
+	"google.golang.org/grpc/metadata"
+)
+
+type Client struct {
+	secret string
+}
+
+func NewClient(
+	secret string,
+) Client {
+	return Client{
+		secret: secret,
+	}
+}
+
+func (c Client) IsAuthorized(ctx context.Context) (bool, error) {
+	ctxMD, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return false, errors.New("no metadata found in context")
+	}
+
+	authorization := ctxMD.Get("authorization")
+	if len(authorization) == 0 {
+		return false, errors.New("token not found in context")
+	}
+
+	tokenStr := strings.TrimPrefix(authorization[0], "Bearer ")
+
+	// decode authorization
+	token, err := jwt.Parse(tokenStr, func(token *jwt.Token) (interface{}, error) {
```

Comment:
Let's talk about this tomorrow! I have messed around with the `dgrijalva` library a bit and I think it exposes a pretty clean way to add custom auth validations by declaring your own implementation of the `jwt.Token` interface. It may not fit the use case here but it also might?

Reply:
Yeah, I looked in a little bit into the `CustomClaims`, but opted to not do it whenever I implemented this because I think I didn't want to implement the interface (so just opted for `MapClaims` 😛 ). I'm open to doing that differently though!

Reply:
That makes total sense!